src/dataset/resource.rs

```diff
@@ -23,15 +23,38 @@ pub enum Type {
     Pdf,
     Csv,
     JsonLd,
+    Xml,
+    Image,
+    Archive,
+    PlainText,
+}
+
+impl From<&'_ str> for Type {
+    fn from(val: &str) -> Self {
+        match val {
+            "text/plain" => Self::PlainText,
+            "application/pdf" => Self::Pdf,
+            "application/xml" => Self::Xml,
+            "image/png" => Self::Image,
+            "wms_xml" => Self::Image,
+            "application/zip" => Self::Archive,
+            "" => Self::Unknown,
```

Comment:
The `_` is a catch-all, i.e. the `""` pattern is not necessary here as it yields the same result. If you want to make it explicit (for future readers) that the empty string also matches `Unknown`, you can merge these branches via

```rust
"" | _ => Self::Unknown
```

but that is a bit redundant.